### PR TITLE
Create new housekeeping function

### PR DESCRIPTION
**Pull request by wagnerd97**

### CODE REVIEW

File: hardware_interface/source/eps.c
Comment (ArashYzd) [inline hardware_interface/source/eps.c:114]: Why not memcpy? because of the lock?

Comment (wagnerd97) [reply hardware_interface/source/eps.c:114]: Memcpy just copies all the data addresses and leaves a shallow copy. This risks the data being changed unexpectedly or deallocated before it can be used. To avoid that, a deep copy is created in this manner.

Comment (ArashYzd) [reply hardware_interface/source/eps.c:114]: Well, the bad news is that these are subject to change, based on what I'm receiving from EPS right now. So, we'll have to rewrite some stuff here. But we can worry about that later.

Comment (wagnerd97) [reply hardware_interface/source/eps.c:114]: Yes. Most of the data from all the subservices are. Thankfully changes can be isolated to a few spots. There's just no generic way to deep copy complex structs so we're stuck with it unless we risk shallow copies. Which we could do. But it brings in other risks such as unexpected NULL pointers

**Review by ArashYzd — APPROVED** [review]
Since my changes are already in the pull request, just wait for that to be merged before merging this. _Also, only Andrew is authorized to merge in this repo._ I guess I was wrong :D

**Pull request merged**
